tests/src/Functional/ContentNewsRenderTest.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_whitelabel\Functional;
+
+use Symfony\Component\DomCrawler\Crawler;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+use Drupal\media\Entity\Media;
+use Drupal\file\Entity\File;
+use Drupal\Tests\TestFileCreationTrait;
+
+/**
+ * Tests that the News content type renders correctly.
+ *
+ * @group batch1
+ */
+class ContentNewsRenderTest extends ContentRenderTestBase {
+
+  use MediaTypeCreationTrait;
+  use TestFileCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'system',
+    'oe_whitelabel_helper',
+    'oe_starter_content_news',
+    'oe_whitelabel_news',
+  ];
+
+  /**
+   * A node to be rendered in diferent display views.
+   *
+   * @var \Drupal\node\Entity\Node
+   */
+  protected $node;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Create a sample image media entity to be embedded.
+    File::create([
+      'uri' => $this->getTestFiles('image')[0]->uri,
+    ])->save();
+    $media_image = Media::create([
+      'bundle' => 'image',
+      'name' => 'Starter Image test',
+      'oe_media_image' => [
+        [
+          'target_id' => 1,
+          'alt' => 'Starter Image test alt',
+          'title' => 'Starter Image test title',
+        ],
+      ],
+    ]);
+    $media_image->save();
+
+    // Create a News node.
+    /** @var \Drupal\node\Entity\Node $node */
+    $node = $this->getStorage('node')->create([
+      'type' => 'oe_news',
+      'title' => 'Test news node',
+      'oe_summary' => 'http://www.example.org is a web page',
+      'body' => 'News body',
+      'oe_publication_date' => [
+        'value' => '2022-02-09T20:00:00',
+      ],
+      'uid' => 1,
+      'status' => 1,
+    ]);
+    $node->set('oe_featured_media', [$media_image]);
+    $node->save();
+    $this->node = $node;
+  }
+
+  /**
+   * Tests that the News page renders correctly in full display.
+   */
+  public function testNewsRenderingFull(): void {
+    $this->drupalGet($this->node->toUrl());
+
+    // Build node full view.
+    $builder = \Drupal::entityTypeManager()->getViewBuilder('node');
+    $build = $builder->view($this->node, 'full');
+    $render = $this->container->get('renderer')->renderRoot($build);
+    $crawler = new Crawler($render->__toString());
```

Comment:
We have a browser test here, so we test the markup rendered browsing the website (line 85), we don't need to render programmatically like in kernel tests (88 to 91).